tutorials/2022-02-09-neow3j-nep11/index.md

```diff
@@ -0,0 +1,647 @@
+            propertiesImageMap.put(tokenId, image);
+        }
+        if (properties.containsKey(propTokenURI)) {
+            String tokenURI = properties.get(propTokenURI);
+            propertiesTokenURIMap.put(tokenId, tokenURI);
+        }
+
+        registryMap.put(tokenId, tokenId);
+        ownerOfMap.put(tokenId, owner.toByteArray());
+        new StorageMap(ctx, createTokensOfPrefix(owner)).put(tokenId, 1);
+
+        increaseBalanceByOne(owner);
+        incrementTotalSupplyByOne();
+        onMint.fire(owner, tokenId, properties);
+    }
+
+    public static boolean burn(ByteString tokenId) {
+        Hash160 owner = ownerOf(tokenId);
+        assert owner != null : "This token id does not exist.";
+        throwIfSignerIsNotOwner(owner);
+
+        registryMap.delete(tokenId);
+        propertiesNameMap.delete(tokenId);
+        propertiesDescriptionMap.delete(tokenId);
+        propertiesImageMap.delete(tokenId);
+        propertiesTokenURIMap.delete(tokenId);
+        ownerOfMap.delete(tokenId);
+
+        new StorageMap(ctx, createTokensOfPrefix(owner)).delete(tokenId);
+        decreaseBalanceByOne(owner);
+        decrementTotalSupplyByOne();
+        return true;
+    }
+```
+
+### Private Helper Methods
+
+Private methods can be used to extract code that is used internally in the contract. The following private methods are used in the NEP-17 example contract. For example, in order to prevent writing the same exception message to check the witness of the owner, the private method `throwIfSignerIsNotContractOwner` allows to write this code only once.
+
+```java
+    private static void throwIfSignerIsNotContractOwner() {
+        assert Runtime.checkWitness(contractOwner) : "No authorization.";
+    }
+
+    private static void throwIfSignerIsNotOwner(Hash160 owner) {
+        assert Runtime.checkWitness(owner) : "No authorization.";
+    }
+
+    private static void increaseBalanceByOne(Hash160 owner) {
+        balanceMap.put(owner.toByteArray(), balanceOf(owner) + 1);
+    }
+
+    private static void decreaseBalanceByOne(Hash160 owner) {
+        balanceMap.put(owner.toByteArray(), balanceOf(owner) - 1);
+    }
+
+    private static void incrementTotalSupplyByOne() {
+        int updatedTotalSupply = contractMap.getInt(totalSupplyKey) + 1;
+        contractMap.put(totalSupplyKey, updatedTotalSupply);
+    }
+
+    private static void decrementTotalSupplyByOne() {
+        int updatedTotalSupply = contractMap.getInt(totalSupplyKey) - 1;
+        contractMap.put(totalSupplyKey, updatedTotalSupply);
+    }
+
+    private static byte[] createTokensOfPrefix(Hash160 owner) {
+        return Helper.concat(tokensOfKey, owner.toByteArray());
+    }
+```
+
+## 5. Compile the Contract
+
+The contract can be compiled using the gradle plugin. First, set the `className` in the file `gradle.build` to the contract's class name. Then, the gradle task `neow3jCompile` can be executed from the project's root path to compile the contract.
```

Comment:
Can we add references to other articles? For example, "As seen in this article, the smart contract is compiled using the Gradle plugin." -- adding the appropriate reference link.

Reply:
I prefer not to add additional links (referring to claude's comment below ;)). Imo, the steps for compiling a contract are explained well enough here.

Reply:
alright. See: https://github.com/neo-project/neo-dev-portal/pull/6#issuecomment-1036035393